src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -419,24 +361,22 @@ public override async Task StopAsync(CancellationToken cancellationToken)
 
             await base.StopAsync(cancellationToken);
             _isHostShuttingDown = true;
-            _loggingScope.Dispose();
+            _loggingScope?.Dispose();
         }
 
         private async Task DeleteTopicSubscriptionAsync(CancellationToken cancellationToken)
         {
-            string rawConnectionString = await GetServiceBusConnectionStringAsync();
-            var serviceBusConnectionProperties = ServiceBusConnectionStringProperties.Parse(rawConnectionString);
-
-            var serviceBusClient = new ServiceBusAdministrationClient(rawConnectionString);
+            ServiceBusAdministrationClient serviceBusClient = await Settings.GetServiceBusAdminClientAsync();
```

Comment:
We already do this on https://github.com/arcus-azure/arcus.messaging/pull/178/files#diff-21920603b21d912cc5678a977a854f57051464f5f3eff50f3dccfcd4d50abca5R210-R215, so maybe move this to a method?

Reply:
Hmm, I don't think that would be much added value as all the arguments that we would be sending to that method are further up. Adding delegation here would make it less readable. The only duplication here is retrieving the information for the SB client and calling that 'subscription exists' method.